### PR TITLE
feat: Adding a detailed readme for HBase replication support.

### PR DESCRIPTION
**Pull request by vermas2012**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Comment by vermas2012** [comment]
@jhambleton PTAL. I had to re-create another PR due to branch issues, I have addressed all the comments except one. https://github.com/googleapis/java-bigtable-hbase/pull/3525#discussion_r815237863

**Comment by vermas2012** [comment]
Updated the PR with some details on authentication. 

@jhambleton should we create a top level section on authentication along with configuration?

File: hbase-migration-tools/bigtable-hbase-replication/README.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+|DeleteFamily|Converts to DeleteFamilyBeforeTimestamp with timestamp=now|See DeleteFamilyBeforeTimestamp|
+|DeleteRow|Converts to DeleteFamilyBeforeTimestamp with timestamp=now for all families|See DeleteFamilyBeforeTimestamp|
+
+The goal of this replication library is to allow migration from HBase to CBT.
+Since CBT will not support these mutations after the users migrate to CBT, they
+are recommended to come up with alternative ways to handle these incompatible
+APIs and not issue them while replication is on.
+
+Another special case is mutations
+with [reverse timestamps](https://hbase.apache.org/book.html#reverse.timestamp).
```
Comment (jhambleton) [inline hbase-migration-tools/bigtable-hbase-replication/README.md:222]: Suggest rephrasing this such as below to specify we're referring to the cell timestamps. Reverse timestamps are commonly used in the row key which may lead users to think they may be impacted by this, when they really aren't.

`Another special case is mutations with custom cell timestamps.`

File: hbase-migration-tools/bigtable-hbase-replication/README.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+HBase uses a `long`
+to store milliseconds while Cloud Bigtable uses `long` to store microseconds.
+This [difference in granularity](https://cloud.google.com/bigtable/docs/hbase-differences#timestamps)
+means, HBase can store 1000 times higher timestamp than Cloud Bigtable. The
+impacted use case is the reverse timestamp, where customers
```
Comment (jhambleton) [inline hbase-migration-tools/bigtable-hbase-replication/README.md:227]: slight edit based on the prior suggestion.

`The impacted use case is custom timestamps, where customers ... `

**Review by jhambleton — CHANGES_REQUESTED** [review]
thank you for incorporating other changes from pr#3525 here! all looks good with minor nit around timestamp incompatibility.

**Review by jhambleton — APPROVED** [review]
thanks for the updates, lgtm

**Review by mutianf — APPROVED** [review]

**Pull request merged**